server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java

```diff
@@ -372,6 +378,48 @@ private void prepareAndValidatePublicationRequest(ProxyContext context, Publicat
         validateRules(publication);
     }
 
+    private void addCustomApplicationRelatedFiles(ProxyContext context, Publication publication) {
+        List<String> existingUrls = publication.getResources().stream()
+                .map(Publication.Resource::getSourceUrl)
+                .toList();
+
+        final String targetFolder;
+            {
```

Comment:
please move the block to a separate method

Reply:
moved